### ISSUE TITLE
Update composer.json with cleentfaar/slack 0.18 +

### ISSUE THREAD

**Issue by LeoBenoist** [cleentfaar/CLSlackBundle#18]
The idea is to be able to update jms to be compatible with Sf3

**Comment by MrNicodemuz** [comment]
+1

**Comment by ChristianVermeulen** [comment]
What I did was simply get rid of this slack bundle and create the ApiClient myself in a SlackService. It is essentially all this bundle does anyway. It's nowhere near perfect, but gets the job done :).

This example is in SF 2.8 But shouldn't be much different in SF3.

It runs:
- CleentFaar\Slack ^0.18.1
- GuzzleHttp ~6.0

```
# config/services.yml

acme.demo_bundle.slack_service:
        class: Acme\DemoBundle\Service\SlackService
        arguments: [%slack_api_token%, @kernel]
````

```
// SlackService.php

namespace Acme\DemoBundle\Service;

use CL\Slack\Payload\ChatPostMessagePayload;
use CL\Slack\Transport\ApiClient;
use Symfony\Component\HttpKernel\Kernel;

class SlackService
{
    /**
     * @var ApiClient
     */
    private $client;

    /**
     * @var Kernel
     */
    private $kernel;

    /**
     * SlackService constructor.
     * @param $token
     * @param Kernel $kernel
     */
    public function __construct($token, Kernel $kernel)
    {
        $this->client = new ApiClient($token);
        $this->kernel = $kernel;
    }


    /**
     * @param string $channel
     * @param string $message
     * @throws \CL\Slack\Exception\SlackException
     */
    public function sendToChannel($channel, $message)
    {
        if ('prod' !== $this->kernel->getEnvironment()) {
            $channel = "#dev-channel";
        }

        $payload = new ChatPostMessagePayload();
        $payload->setChannel($channel);
        $payload->setText($message);
        $payload->setUsername('BotName');
        $payload->setIconEmoji('Emoji');

        $this->client->send($payload);

        return;
    }
}
```

```
$this->get('acme.demo_bundle.slack_service')->sendToChannel('#general', 'message');
```

Don't forget to load your config in the compiler pass!

@cleentfaar I guess the simpelest is to just make sure this bundle requires your 'slack' repo at version 0.18.1?

**Comment by MrNicodemuz** [comment]
I migrated to this bundle:
https://github.com/DZunke/SlackBundle

It supports Symfony 3.

**Comment by cleentfaar** [comment]
First of all, sorry for the late reply and thanks for reporting! 

The issue has been fixed today with release 0.20 (link: https://github.com/cleentfaar/CLSlackBundle/releases/tag/0.20.1). Unfortunately I have had to raise the minimal required PHP version to 5.5 to support all different package versions.

Both the bundle and the library now support Symfony 2.3+ **and** 3.0+.

If any problem arises from updating, please create a new issue for it. Thanks again!

**Comment by LeoBenoist** [comment]
Thanks :)
